datmo/cli/command/task.py

```diff
@@ -47,15 +48,14 @@ def run(self, **kwargs):
         self.cli_helper.echo(__("info", "cli.task.run"))
 
         # Create input dictionaries
-        snapshot_dict = {
-            "environment_definition_filepath":
+        snapshot_dict = {}
+        if kwargs['environment_definition_filepath']:
+            snapshot_dict["environment_definition_filepath"] =\
                 kwargs['environment_definition_filepath']
-        }
-
         if not isinstance(kwargs['cmd'], list):
             if platform.system() == "Windows":
                 kwargs['cmd'] = kwargs['cmd']
-            else:
+            elif type(kwargs['cmd']) is str:
```

Comment:
@shabazpatel for consistency with python2 and python3 we should not use "type(param)" instead we should only use "isinstance(obj, type)" so in this case it would be `isinstance(kwargs['cmd'], str)`